src/main/java/org/eclipse/golo/runtime/AugmentationMethodFinder.java

```diff
@@ -129,7 +132,7 @@ public MethodHandle find(Class<?> definingModule, Class<?> augmentedClass) {
     protected String augmentationClassName(Class<?> definingModule, String augmentationName) {
       int idx = augmentationName.lastIndexOf(".");
       if (idx == -1) { return augmentationName; }
-      return (new StringBuilder(augmentationName)).replace(idx, idx+1, "$").toString();
+      return (new StringBuilder(augmentationName)).replace(idx, idx + 1, "$").toString();
```

Comment:
But since you know the index, you can replace with a string concatenation, which ends up being a `StringBuilder`. So yes, it’s better with `StringBuilder` as you did.

Note that JDK9 will use `invokedynamic` on some string concatenation call sites, to dispatch to optimized variants. Weird and funny stuff, isn’t it? :smile:

See http://openjdk.java.net/jeps/280